### ISSUE TITLE
Client 1.16.3 Crash When putting entitie in Cage

### ISSUE THREAD

**Issue by MachinegunMarty** [Corgam/CagedMobs#6]
[latest.log](https://github.com/Corgam/CagedMobs/files/5506924/latest.log)

I have a MC 1.16.3 Forge Server and 2 Forge clients who connect.
Using your latest version: cagedmobs-1.16.4-1.2.1.jar

Sampled a Minecraft: cow, put hay in the cage, right clicked with the sample and my client crashed.
Restarted MC and tried to log in again, crashed immediately.
Had optifine G5 and Latest OptiForge installed and thought that was the problem. Removed Optifine and OptiForge from client, Log in still crashes the client.

It's like i'm stuck in the GUI of the cage. Every time i log in i see i'm still in the GUI and then crashes.

I did it (player 1), but player 2 also can't log in anymore. So can't log into the server anymore.

If can supply any extra information please let me know.

grtz,

Machinegun Marty

**Comment by Corgam** [comment]
I'm really sorry, I think there is a problem with rendering the entity model inside the cage. I will try to fix it asap, for now you can use third party software to edit the save and remove the cage block or use backup of your server if you have one.

EDIT: both players were next to the cage when they crashed, right?

**Comment by MachinegunMarty** [comment]
Hey Corgam,

Np man. I will wait on the fix and test it for you.

I got backups.

The crashed happened by right clicking the cage and player 2 was nowhere near it. After i crashed player 2 got kicked to.
Strange thing is: Server just runs on and doesn't crash, just the clients crash.

If i can test anything for you just holla ;-)

grtz,

Machinegun Marty

**Comment by Corgam** [comment]
I hope I fixed it, please try it again when new version will come out. And reopen this issue or write me a PM if it worked.

**Comment by MachinegunMarty** [comment]
Hey Corgam,

Downloaded v1.2.2 and installed it on my crashing world and it works like a charm !

I loaded in without the crashing problems and the cage is still there.

The hopper cage is/ was empty (only hay on the ground) but there was leather and meat in the box.

So to test: i threw away the old cage and made 2 new ones: a normal one and a hopper cage.

1. Put hay in both cages and filled 2x syringe with minecraft: cow again and right clicked the cage.
2. The syringe empties in both cages, but in both cages no entity shows.
3. Right clicking the normal cage gave me leather and meat
4. Looking in the Minecraft: chest under the hopper cage also produced leather and meat.

So the cages are working now, but they don't show the entities inside the cage.

First tested without optifine (which worked) and after that with optifine (which also worked) just to rule out that optifine is not showing the entities in the cages.

Thanx for the quick fix !

It's now working again (with and without optifine), and the entities are in the cage and are also working, but the entities don't show when looking at the cage.

You want me to make a new Issue from this on your tracker ?

grtz,

Machinegun Marty
